internal/services/v1/lookupwatch.go

```diff
@@ -0,0 +1,267 @@
+		}
+	}
+
+	usagemetrics.SetInContext(ctx, &dispatchv1.ResponseMeta{
+		DispatchCount: 1,
+	})
+
+	updates, errchan := ds.Watch(ctx, afterRevision)
+
+	for {
+		select {
+		case update, ok := <-updates:
+			if ok {
+				//TODO When do we CLOSE STREAM ?
+				return lw.processWatchResponse(&ctx, req, update, &stream)
+			}
+		case err := <-errchan:
+			switch {
+			case errors.As(err, &datastore.ErrWatchCanceled{}):
+				return status.Errorf(codes.Canceled, "watch canceled by user: %s", err)
+			case errors.As(err, &datastore.ErrWatchDisconnected{}):
+				return status.Errorf(codes.ResourceExhausted, "watch disconnected: %s", err)
+			default:
+				return status.Errorf(codes.Internal, "watch error: %s", err)
+			}
+		}
+	}
+}
+
+// Called at each notification of the Wattch API
+func (lw *lookupWatchServer) processWatchResponse(
+	ctx *context.Context,
+	req *v1lookupwatch.WatchAccessibleResourcesRequest,
+	updates *datastore.RevisionChanges,
+	stream *v1lookupwatch.LookupWatchService_WatchAccessibleResourcesServer,
+) error {
+	for _, update := range updates.Changes {
+		if err := lw.processUpdate(ctx, req, update, &updates.Revision, stream); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Called every time a relation is changed in spicedb
+func (lw *lookupWatchServer) processUpdate(
+	ctx *context.Context,
+	req *v1lookupwatch.WatchAccessibleResourcesRequest,
+	update *core.RelationTupleUpdate,
+	atRevision *datastore.Revision,
+	stream *v1lookupwatch.LookupWatchService_WatchAccessibleResourcesServer,
+) error {
+
+	//CALL LOOKUPSUBJECTS
+	var subjects []string
+	if req.SubjectObjectType == update.Tuple.Subject.Namespace {
+		// TODO: we'll revisit that once we have arrow resolution, we leave as it is meanwhile for our testcases
+		subjects = append(subjects, update.Tuple.Subject.ObjectId)
+	} else if update.Tuple.Subject.Relation == "" {
+		return status.Errorf(
+			codes.Internal,
+			"cannot handle the following relation, arrow resolution not implemented: %s:%s",
+			update.Tuple.Subject.Namespace, update.Tuple.Subject.ObjectId,
+		)
+	} else {
+		lsStream := dispatchpkg.NewHandlingDispatchStream(*ctx, func(result *dispatchv1.DispatchLookupSubjectsResponse) error {
+			for _, subject := range result.FoundSubjects {
+				subjects = append(subjects, subject.SubjectId)
+			}
+			return nil
+		})
+		err := lw.dispatch.DispatchLookupSubjects(
+			&dispatchv1.DispatchLookupSubjectsRequest{
+				Metadata: &dispatchv1.ResolverMeta{
+					AtRevision:     atRevision.String(),
+					DepthRemaining: lw.defaultDepth,
+				},
+				ResourceIds: []string{update.Tuple.Subject.ObjectId},
+				ResourceRelation: &core.RelationReference{
+					Namespace: update.Tuple.Subject.Namespace,
+					// TODO: arrow resolution, get Permission from @joey's code
+					Relation: update.Tuple.ResourceAndRelation.Relation,
+				},
+				SubjectRelation: &core.RelationReference{
+					Namespace: req.SubjectObjectType,
+					// TODO: What value to set ?
```

Comment:
the Relation from the specified subject type in the API call

Reply:
I replaced this parameter to LS API with :

```
		SubjectRelation: &core.RelationReference{
			Namespace: req.SubjectObjectType,
			Relation:  req.OptionalSubjectRelation,
		},
```

Not sure if it's ok for you (if the user don't supply any req.OptionalSubjectRelation, I'll get empty string value, is it allright ?)

Reply:
You'll need to convert empty string to `tuple.Ellipsis` since this is going through the dispatcher.

Use: `stringz.DefaultEmpty(req.OptionalSubjectRelation, tuple.Ellipsis)`

Reply:
done, thanks !